jupyterhub/alembic/versions/4621fec11365_manage_roles.py

```diff
@@ -0,0 +1,40 @@
+"""manage roles
+
+Revision ID: 4621fec11365
+Revises: 3c2384c5aae1
+Create Date: 2024-04-09 09:28:13.576059
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '4621fec11365'
+down_revision = '3c2384c5aae1'
+branch_labels = None
+depends_on = None
+
+import sqlalchemy as sa
+from alembic import op
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column(
```

Comment:
in case of a long upgrade, these tables may not have been created yet. You can check for their existence before altering them, as done [here](https://github.com/jupyterhub/jupyterhub/blob/f9fb650a7b1cfbc5306cb674ec3038f1801f4e17/jupyterhub/alembic/versions/56cc5a70207e_token_tracking.py#L33):

```python
    engine = op.get_bind().engine
    tables = sa.inspect(engine).get_table_names()
    if table_name in tables:
        op.add_column(table_name, ...)
```

If the tables don't exist yet, there's no upgrade to do because they will be created on first start.